flink/v1.15/flink/src/test/java/org/apache/iceberg/flink/sink/TestPartitionUdf.java

```diff
@@ -0,0 +1,224 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.flink.sink;
+
+import java.math.BigDecimal;
+import java.nio.ByteBuffer;
+import java.util.List;
+import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
+import org.apache.flink.table.api.TableEnvironment;
+import org.apache.flink.table.api.TableResult;
+import org.apache.flink.table.api.bridge.java.StreamTableEnvironment;
+import org.apache.flink.types.Row;
+import org.apache.flink.util.CloseableIterator;
+import org.apache.iceberg.expressions.Literal;
+import org.apache.iceberg.relocated.com.google.common.collect.Lists;
+import org.apache.iceberg.transforms.Transform;
+import org.apache.iceberg.transforms.Transforms;
+import org.apache.iceberg.types.Types;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class TestPartitionUdf {
+
+  public static TableEnvironment tEnv;
+
+  @BeforeClass
+  public static void before() {
+    StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+    tEnv = StreamTableEnvironment.create(env);
+    tEnv.createTemporarySystemFunction("buckets", PartitionTransformUdf.Bucket.class);
+    tEnv.createTemporarySystemFunction("truncates", PartitionTransformUdf.Truncate.class);
+  }
+
+  protected List<Row> sql(String query, Object... args) {
+    TableResult tableResult = tEnv.executeSql(String.format(query, args));
+    try (CloseableIterator<Row> it = tableResult.collect()) {
+      return Lists.newArrayList(it);
+    } catch (Exception e) {
+      throw new RuntimeException("Failed to collect table result", e);
+    }
+  }
+
+
+  @Test
+  public void testBucketDate() {
+    Transform<Object, Integer> bucket = Transforms.bucket(Types.DateType.get(), 4);
+
+    String date = "2022-05-20";
+    Integer value = bucket.apply(Literal.of(date).to(Types.DateType.get()).value());
+    String expectHumanString = bucket.toHumanString(value);
+
+    List<Row> sql = sql("SELECT buckets(4, DATE '%s')", date);
+    Assert.assertEquals(expectHumanString, sql.get(0).getField(0));
+  }
+
+  @Test
+  public void testBucketInt() {
+    Transform<Object, Integer> bucket = Transforms.bucket(Types.IntegerType.get(), 4);
+
+    int num = 10;
+    Integer obj = bucket.apply(num);
+    String expectHumanString = bucket.toHumanString(obj);
+
+    List<Row> sql = sql("SELECT buckets(4, %d)", num);
+    Assert.assertEquals(expectHumanString, sql.get(0).getField(0));
+  }
+
+  @Test
+  public void testBucketLong() {
+    Transform<Object, Integer> bucket = Transforms.bucket(Types.LongType.get(), 4);
+
+    long num = 10;
+    Integer obj = bucket.apply(num);
+    String expectHumanString = bucket.toHumanString(obj);
```

Comment:
Nit: Can we have a test where the expected human string is tested? Eg with a hard coded string value as the expected human string and then the actual value is the result of this call?

Reply:
Looks good, it will be clearer. I will modify it.